### PR TITLE
Rewrite React Hooks in TypeScript

### PR DESCRIPTION
**Pull request by axbarsan**
Towards https://github.com/giantswarm/giantswarm/issues/13001

Again, seems big due to import changes.

This PR:
* Refactors React Hooks to TypeScript.
* Updates `prettier` to `2.1.1` to support labeled tuples in TypeScript 4.0
* Rename the `effects` folder to `hooks`

### CODE REVIEW

File: src/lib/hooks/useCopyToClipboard.ts [hunk 1]
```diff
@@ -0,0 +1,24 @@
+import copy from 'copy-to-clipboard';
+import { useEffect, useState } from 'react';
+
+/**
+ * Hook for copying text to the user's clipboard.
+ *
+ * Note: Set value to `null` for resetting the clipboard state.
+ */
+function useCopyToClipboard(): [
+  hasContentInClipboard: boolean,
```
Comment (axbarsan) [inline src/lib/hooks/useCopyToClipboard.ts:10]: Labeled tuple types in TypeScript 4.0

**Review by ubergesundheit — APPROVED** [review]
I like the renaming from effects to hooks :+1: 

Maybe you could also move all hooks into the same place?

Here are some more: https://github.com/giantswarm/happa/tree/master/src/hooks ?

**Comment by axbarsan** [comment]
> Maybe you could also move all hooks into the same place?
> 
> Here are some more: [`master`/src/hooks](https://github.com/giantswarm/happa/tree/master/src/hooks) ?

Yes! Good catch, I'll do that in this PR

**Review by ubergesundheit — APPROVED** [review]
:+1:

**Pull request merged**